### PR TITLE
DATAUP-497: job comm channel 

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

- split out JobCommMessages into a separate module
- add validation tests for run_status and a general message validator for incoming kernel messages

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-497
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: docs/design/job_architecture.md [hunk 1]
```diff
@@ -18,13 +18,11 @@ The narrative backend uses the JobComm module to send and receive messages from
 The front- and backend have a shared vocabulary of message types and parameters, loaded from the file `kbase-extension/static/kbase/config/job_config.json`. In the frontend, the message param and type names are stored in the JobCommMessages object exported by `jobCommChannel.js`:
 
 ```js
-define(['common/jobCommChannel'], (JobComms) => {
+define(['common/jobCommMessages'], (JobCommMessages) => {
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:21]: Throughout the PR, imports of `jobCommChannel` have been switched for importing the `jobCommMessages` module instead.

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 2]
```diff
@@ -37,43 +37,14 @@ define([
...
-        // message types
-        MESSAGE_TYPE: JobConfig.message_types,
-        RESPONSES,
-        REQUESTS,
-        // standardised params
-        PARAM,
-        // channel types
-        CHANNEL,
-    };
+        { CHANNEL, PARAM, REQUESTS, RESPONSES } = JobCommMessages;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:47]: Moved `JobCommMessages` into its own file with some basic tests.

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 3]
```diff
@@ -293,54 +264,23 @@ define([
...
-
                 case RESPONSES.LOGS:
-                    Object.keys(msgData).forEach((_jobId) => {
-                        const jobData = msgData[_jobId];
-                        if (this.validationFn[msgType](jobData)) {
-                            this.sendBusMessage(CHANNEL.JOB, _jobId, RESPONSES.LOGS, jobData);
-                        } else {
-                            this.reportCommMessageError({ msgType, msgData: jobData });
-                        }
-                    });
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:314]: factor out the repeated pattern

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 4]
```diff
@@ -104,6 +105,91 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/ui', 'uti
...
 
+    /**
+     *
+     * @param {object} args with keys
+     *  message - the message for validation
+     *  type    - the message type (one of jcm.MESSAGE_TYPEs)
+     *
+     * @returns {boolean}
+     */
+    function validateMessage(args) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:116]: New: function to validate a whole message as would be received from the kernel

File: docs/design/job_architecture.md [hunk 5]
```diff
@@ -18,11 +18,9 @@ The narrative backend uses the JobComm module to send and receive messages from
 The front- and backend have a shared vocabulary of message types and parameters, loaded from the file `kbase-extension/static/kbase/config/job_config.json`. In the frontend, the message param and type names are stored in the JobCommMessages object exported by `jobCommChannel.js`:
 
 ```js
-define(['common/jobCommChannel'], (JobComms) => {
+define(['common/jobCommMessages'], (jcm) => {
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:21]: Throughout the PR, imports of jobCommChannel have been switched for importing the jobCommMessages module as `jcm` and removing the line `const jcm = JobComms.JobCommMessages`.

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 6]
```diff
@@ -37,43 +37,14 @@ define([
...
-        // message types
-        MESSAGE_TYPE: JobConfig.message_types,
-        RESPONSES,
-        REQUESTS,
-        // standardised params
-        PARAM,
-        // channel types
-        CHANNEL,
-    };
+        { CHANNEL, PARAM, REQUESTS, RESPONSES } = jcm;
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:47]: JobCommMessages is now in its own module. Hurrah!

File: test/unit/spec/common/jobCommMessages-spec.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+define(['common/jobCommMessages'], (JobCommMessages) => {
+    'use strict';
+
+    describe('the JobCommMessages module', () => {
+        const expectedKeys = ['CHANNEL', 'MESSAGE_TYPE', 'PARAM', 'REQUESTS', 'RESPONSES'];
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommMessages-spec.js:5]: Some basic tests for the JobCommMessages module that will also check the integrity of the job_config.json file.

**Review by briehl — APPROVED** [review]
I'm a fan of decomposition!

**Pull request merged**